.github/workflows/check-in-tree-build.yml

```diff
@@ -66,6 +66,12 @@ jobs:
           repository: llvm/llvm-project
           ref: main
           path: llvm-project
+      - name:  Checkout SPIR-V Headers
```

Comment:
Is this needed actually, or would it be done automatically by cmake for in-tree builds?

Reply:
The idea behind the manual installation for GH Actions has originally come from @AlexeySachkov, in particular, aiming to minimize the risk of proxy issues during configure time (as different workflow steps are being performed in separate, newly configured shells). On the other hand, we end up _not_ testing the configure-time download in CI, yet I'm not sure which workflow to base such testing off - creating a separate check sounds like an overkill to me.

Reply:
I believe these in-tree actions are already testing the configure-time download anyway, because you do not specify any SPIRV-Headers path to cmake for in-tree builds currently.  So the checkout done here is not used, unless I am missing something?

For example the [Windows (Release) CI log](https://github.com/KhronosGroup/SPIRV-LLVM-Translator/pull/813/checks?check_run_id=3054935392)  from your latest run seems to confirm my suspicion:
```
 -- SPIR-V Headers location is not specified. Will try to download
          spirv.hpp from https://github.com/KhronosGroup/SPIRV-Headers into
          D:/a/SPIRV-LLVM-Translator/SPIRV-LLVM-Translator/build/projects/SPIRV-LLVM-Translator/SPIRV-Headers
Microsoft (R) Build Engine version 16.10.2+857e5a733 for .NET Framework
Copyright (C) Microsoft Corporation. All rights reserved.

  Checking Build System
  Creating directories for 'spirv-headers-populate'
  Performing download step (git clone) for 'spirv-headers-populate'
  Cloning into 'SPIRV-Headers'...
```

I don't have a strong preference if we do the download during configure-time or separately upfront; but I'd rather not do extra clones that aren't used. ;-)

Reply:
The in-tree build part seems to be a bug - I was intending to make it a non-download workflow as well. I'll look into this shortly; meanwhile, the clang-format action should actually suffice for testing the configure-time download.

Reply:
Never mind the Windows & MacOS thing - as pointed out by @AlexeySachkov, it was just a matter of updating the corresponding workflows. Fixed now.

Reply:
Let's compare timings now to see if separate checkout through corresponding GitHub Actions is more efficient that checkout through CMake or not. We had 2:02 for configure and let's say 0 for checkout

Reply:
And now we have 2:34 + 0:04. It is hard to compare performance on Windows on some virtual HW in the cloud :)

Reply:
Yep, I'm also suspecting that we can't hope for consistency with a vastly loaded cloud server.